### PR TITLE
Container adapters

### PR DESCRIPTION
**Pull request by eugeneko**

### CODE REVIEW

**Comment by rokups** [comment]
@JTippetts @ArnisLielturks you may be interested in testing this.

File: CMake/Modules/UrhoOptions.cmake [hunk 1]
```diff
@@ -124,6 +125,10 @@ if (URHO3D_CSHARP)
     set (URHO3D_MONOLITHIC_HEADER ON)   # Used by wrapper code
 endif ()
 
+if (URHO3D_CONTAINER_ADAPTERS)
+    set (URHO3D_CONTAINER_ADAPTERS ON)
```
Comment (rokups) [inline CMake/Modules/UrhoOptions.cmake]: What did you try to do here? It is essentially a noop.

File: Source/ThirdParty/EASTL/include/EASTL/string.h [hunk 2]
```diff
@@ -934,6 +887,111 @@ namespace eastl
...
 		}
 
+#ifdef URHO3D_CONTAINER_ADAPTERS
+		using Iterator = iterator;
+		using ConstIterator = const_iterator;
+
+		basic_string(value_type x, size_type n) : basic_string(n, x)
+		{
+			static_assert(0, "This constructor is deprecated. "
+				"Change the order of elements from string(char, count) to string(count, char).");
```
Comment (eugeneko) [inline Source/ThirdParty/EASTL/include/EASTL/string.h]: TODO: -> "order of arguments"

**Comment by ArnisLielturks** [comment]
ooo, nice! Will test it when and if I have a chance

File: Source/ThirdParty/EASTL/include/EASTL/string.h [hunk 3]
```diff
@@ -934,6 +887,111 @@ namespace eastl
...
+		void Resize(size_type size) { resize(size); }
+		void Resize(size_type size, value_type value) { resize(size, value); }
+		void Clear() { clear(); }
+		void Compact() { shrink_to_fit(); }
+		void Swap(this_type& other) { swap(other); }
+
+    	this_type Substring(size_type position, size_type n = npos) const { return substr(position, n); }
+		this_type Trimmed() const { return trimmed(); }
+		this_type ToUpper() const { return to_upper(); }
+		this_type ToLower() const { return to_lower(); }
```
Comment (rokups) [inline Source/ThirdParty/EASTL/include/EASTL/string.h]: Tabs and spaces are mixed all over \o/

Comment (eugeneko) [reply Source/ThirdParty/EASTL/include/EASTL/string.h]: Oh shit. I totally forgot about it. Sometimes I copy-pasted code from headers, sometimes I worked in-place.

**Pull request merged**